libnavigation-ui/src/main/java/com/mapbox/navigation/ui/internal/building/BuildingLayerSupport.kt

```diff
@@ -0,0 +1,99 @@
+package com.mapbox.navigation.ui.internal.building
+
+import android.graphics.Color
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.style.expressions.Expression
+import com.mapbox.mapboxsdk.style.expressions.Expression.all
+import com.mapbox.mapboxsdk.style.expressions.Expression.distance
+import com.mapbox.mapboxsdk.style.expressions.Expression.eq
+import com.mapbox.mapboxsdk.style.expressions.Expression.get
+import com.mapbox.mapboxsdk.style.expressions.Expression.id
+import com.mapbox.mapboxsdk.style.expressions.Expression.literal
+import com.mapbox.mapboxsdk.style.expressions.Expression.lt
+import com.mapbox.mapboxsdk.style.layers.Layer
+import com.mapbox.mapboxsdk.style.layers.PropertyValue
+
+/**
+ * This class handles shared code for the [BuildingExtrusionHighlightLayer]
+ * and [BuildingFootprintHighlightLayer] classes which show
+ * [com.mapbox.mapboxsdk.style.layers.FillExtrusionLayer] 3D buildings
+ * and [com.mapbox.mapboxsdk.style.layers.FillLayer] polygons, respectively.
+ * For now, this class is only compatible with the Mapbox Streets v7 vector tile source
+ * because that's what the default Navigation UI SDK styles use.
+ */
+internal class BuildingLayerSupport {
+
+    /**
+     * Gets the correct [Expression.all] expression to show the building
+     * extrusion associated with the query [LatLng].
+     *
+     * @param queryLatLng the [LatLng] to use in determining the building ID to use in the
+     * expression and which building is closest to the coordinate.
+     * @return an [Expression.all] expression
+     */
+    fun getBuildingFilterExpression(queryLatLng: LatLng, mapboxMap: MapboxMap): Expression {
```

Comment:
Instead of MapboxMap being a dependency why not the result of getBuildingId() instead? That would seem to be more accurate. Your doc. for getBuildingId() even mentions that the result of getBuildingId() is used for this method.